client_query_limits_test.go

```diff
@@ -0,0 +1,84 @@
+package fauna_test
+
+import (
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/fauna/fauna-go"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestClientRetriesWithQueryLimits(t *testing.T) {
+	t.Setenv(fauna.EnvFaunaSecret, "secret")
+	dbName := os.Getenv("QUERY_LIMITS_DB")
+	collName := os.Getenv("QUERY_LIMITS_COLL")
```

Comment:
suggest using `os.LookupEnv` and if not found then using `t.Skip()` -- probably should do the same for `FAUNA_ENDPOINT` and `FAUNA_SECRET` as I believe those are required for this test to run.